pumpkin/src/client/container.rs

```diff
@@ -27,78 +39,405 @@ impl Player {
+                    self.double_click(opened_container.as_deref_mut(), slot)
+                } else {
+                    Err(InventoryError::InvalidPacket)
+                }
+            }
+            ClickType::MouseDrag { drag_state } => {
+                if drag_state == MouseDragState::End {
+                    update_whole_container = true;
+                }
+                self.mouse_drag(drag_handler, opened_container.as_deref_mut(), drag_state)
+            }
+            ClickType::DropType(_drop_type) => todo!(),
+        }?;
+        if let Some(mut opened_container) = opened_container {
+            if update_whole_container {
+                drop(opened_container);
+                self.send_whole_container_change(server).await?;
+            } else if let container_click::Slot::Normal(slot_index) = click.slot {
+                let combined_container = OptionallyCombinedContainer::new(
+                    &mut self.inventory,
+                    Some(&mut opened_container),
+                );
+                if let Some(slot) = combined_container.get_slot_excluding_inventory(slot_index) {
+                    let slot = Slot::from(slot);
+                    drop(opened_container);
+                    self.send_container_changes(server, slot_index, slot)
+                        .await?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn mouse_click(
+        &mut self,
+        opened_container: Option<&mut Box<dyn Container>>,
+        mouse_click: MouseClick,
+        slot: container_click::Slot,
+    ) -> Result<(), InventoryError> {
+        let mut container = OptionallyCombinedContainer::new(&mut self.inventory, opened_container);
+
+        match slot {
+            container_click::Slot::Normal(slot) => {
+                container.handle_item_change(&mut self.carried_item, slot, mouse_click)
+            }
+            container_click::Slot::OutsideInventory => Ok(()),
+        }
+    }
+
+    fn shift_mouse_click(
+        &mut self,
+        opened_container: Option<&mut Box<dyn Container>>,
+        slot: container_click::Slot,
+    ) -> Result<(), InventoryError> {
+        let mut container = OptionallyCombinedContainer::new(&mut self.inventory, opened_container);
+
+        match slot {
+            container_click::Slot::Normal(slot) => {
+                let all_slots = container.all_slots();
+                if let Some(item_in_pressed_slot) = all_slots[slot].to_owned() {
+                    let slots = all_slots.into_iter().enumerate();
+                    // Hotbar
+                    let find_condition = |(slot_number, slot): (usize, &mut Option<ItemStack>)| {
+                        // TODO: Check for max item count here
+                        match slot {
+                            Some(item) => {
+                                if item.item_id == item_in_pressed_slot.item_id
+                                    && item.item_count != 64
```

Comment:
Stack sized does varie. We parse it from items.json

Reply:
Yes, I thought i put a TODO in the places where needed 👍